sdk/python/kfp/dsl/_container_op.py

```diff
@@ -457,6 +458,31 @@ def add_volume_devices(self, volume_device) -> 'Container':
                                                 volume_device)
         return self
 
+    def set_env_variable(self, name: str, value: str) -> 'Container':
+        """Sets environment variable to the container (v2 only).
+
+        Args:
+          name: The name of the environment variable.
+          value: The value of the environment variable.
+        """
+
+        if not kfp.COMPILING_FOR_V2:
+            raise ValueError(
+                'set_env_variable is v2 only. Use add_env_variable for v1.')
+
+        # Merge with any existing environment varaibles
+        self.env_dict = {
+            env.name: env.value for env in self._container_spec.env or []
+        }
+        self.env_dict[name] = value
+
+        del self._container_spec.env[:]
+        self._container_spec.env.extend([
+            _PipelineContainerSpec.EnvVar(name=name, value=value)
+            for name, value in self.env_dict.items()
+        ])
+        return self
```

Comment:
Maybe not for this PR, but I realized a lot of setting/adding functions return the Container. Wondering whether returning the container is needed, since the user setting the requirements already has the container?

Reply:
Yes, the reason for returning container is to allow chained setting, for example, `task.set_cpu_limi(...).set_env_variable(...).set_caching_option(...)`. 
Without `return self`, the above setting has to be split into multi-lines:
```
task.set_cpu_limi(...)
task.set_env_variable(...)
task.set_caching_option(...)
```